src/components/movie-card/movie-card.jsx

```diff
@@ -1,14 +1,85 @@
 import PropTypes from "prop-types";
 import { Button, Card } from "react-bootstrap";
+import { Link } from "react-router-dom";
+import { useState } from "react";
+
+export const MovieCard = ({ movie, onToggleFavorite }) => {
+  const token = localStorage.getItem("token");
+  const [isFavorite, setIsFavorite] = useState(false);
+  const storedUser = JSON.parse(localStorage.getItem("user"));
+  const [user, setUser] = useState(storedUser ? storedUser : null);
+
+  const handleToggleFavorite = () => {
+    setIsFavorite(!isFavorite);
+    onToggleFavorite(movie._id, !isFavorite);
+
+    if (isFavorite) {
+      removeFav();
+    } else {
+      addFav();
+    }
+  };
+
+  const checkFavorite = () => {
+    if (user.FavoriteMovies.includes(movie._id)) {
+      setIsFavorite(true);
+    }
+  };
+
+  const addFav = () => {
+    fetch(
+      `https://radiant-river-68463-0f7c4a72bc48.herokuapp.com/users/${user._id}/movies/${movie._id}`,
+      {
+        method: "PATCH",
+        headers: {
+          Authorization: `Bearer ${token}`,
+          "Content-Type": "application/json",
+        },
+      }
+    )
+      .then((response) => response.json())
+      .then((userDetails) => {
+        // update user in local storage
+        setUser(userDetails);
+        console.log("-----user---after--", user);
+        alert("Movie added to your favourites");
+      })
+      .catch((e) => console.log(e));
+  };
+
+  const removeFav = () => {
+    fetch(
+      `https://radiant-river-68463-0f7c4a72bc48.herokuapp.com/users/${user._id}/favorites/${movie._id}`,
+      {
+        method: "DELETE",
+        headers: {
+          Authorization: `Bearer ${token}`,
+          "Content-Type": "application/json",
+        },
+      }
+    )
+      .then((response) => response.json())
+      .then((userDetails) => {
+        console.log("-----user---before--", user);
+        console.log("-----userDetails---removeFav--", userDetails);
```

Comment:
Kindly avoid submitting code review while leaving commented code or code used for debugging purposes.

Reply:
done